components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java

```diff
@@ -4383,4 +4383,22 @@ public static String resolveTenantDomain(HttpServletRequest request) {
         }
         return IdentityTenantUtil.getTenantDomainFromContext();
     }
+
+    /**
+     * Check whether introspecting tokens from different tenants is allowed.
+     *
+     * @return true/false Allow or block cross tenant token inspection.
+     */
+    public static boolean isCrossTenantTokenInspectionAllowed() {
+
+        String allowCrossTenantTokenInspectionConfigValue = IdentityUtil.getProperty(
```

Comment:
We use OAuthServerConfiguration class to read the configs from identity.xml file. Can you move the config retrieval logic there?